pkg/collector/parser/receiver.go

```diff
@@ -155,9 +155,21 @@ func portName(receiverName string, port int32) string {
 }
 
 func portFromEndpoint(endpoint string) (int32, error) {
-	i := strings.LastIndex(endpoint, ":") + 1
-	part := endpoint[i:]
-	port, err := strconv.ParseInt(part, 10, 32)
+	var err error
+	var cport string
+	var port int64
+
+	var r = regexp.MustCompile(":[0-9]+")
+
+	if r.MatchString(endpoint) {
+		cport = strings.Replace(r.FindString(endpoint), ":", "", -1)
+	}
+	port, err = strconv.ParseInt(cport, 10, 32)
```

Comment:
this looks weird. If there is no match the `strconv.ParseInt` is called on an empty `cport`?

Reply:
it is more idiomatic to directly return `0, nil`  if the match is false, if this is what you want to return

Reply:
@pavolloffay , I did it to get the `strconv.ParseInt` error return.